app/controllers/search_controller.rb

```diff
@@ -1,8 +1,21 @@
 class SearchController < ApplicationController
   def search
+    default_owner_search_params = {highlight: {fields: [:nickname, :name, :company, :blog]}, page: params[:page], per_page: 10}
+    default_scraper_search_params = {fields: [{full_name: :word_middle}, :description, {scraped_domain_names: :word_end}], highlight: true, page: params[:page], per_page: 10}
+
     @q = params[:q]
-    @scrapers = Scraper.search @q, fields: [{full_name: :word_middle}, :description, {scraped_domain_names: :word_end}], highlight: true, page: params[:page], per_page: 10
-    @owners = Owner.search @q, highlight: {fields: [:nickname, :name, :company, :blog]}, page: params[:page], per_page: 10
     @type = params[:type]
+    @show = params[:show]
+
+    @owners = Owner.search @q, default_owner_search_params
+
+    all_scrapers = Scraper.search @q, default_scraper_search_params
+    @unfiltered_scrapers_count = all_scrapers.total_count
+
+    if @show == "all" && @type != "users"
```

Comment:
You shouldn't actually need to test what the @type is here, right? Okay, it makes the code very slightly less efficient but it also makes it slightly more complicated